### PR TITLE
[DNM] Debug clippy

### PR DESCRIPTION
**Pull request by alvicsam**

### CODE REVIEW

**Comment by ggwpez** [comment]
If this goes green we may try to run `cargo +1.70 clippy --all-targets --locked -p polkadot-runtime-parachains` directly in CI to check that this fails with the expected 159 errors.

File: scripts/ci/gitlab/pipeline/test.yml [hunk 1]
```diff
@@ -114,5 +114,6 @@ cargo-clippy:
     - .docker-env
     - .test-refs
   script:
+    - echo $RUSTFLAGS
     - cargo version && cargo clippy --version
     - SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo clippy --locked --all-targets
```
Comment (ggwpez) [inline scripts/ci/gitlab/pipeline/test.yml:119]:
```suggestion
    - SKIP_WASM_BUILD=1 cargo +1.70 clippy --all-targets --locked -p polkadot-runtime-parachains
```

File: scripts/ci/gitlab/pipeline/test.yml [hunk 2]
```diff
@@ -114,5 +114,6 @@ cargo-clippy:
     - .docker-env
     - .test-refs
   script:
+    - echo $RUSTFLAGS
     - cargo version && cargo clippy --version
-    - SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo clippy --locked --all-targets
+    - SKIP_WASM_BUILD=1 cargo +1.70 clippy --all-targets --locked -p polkadot-runtime-parachains
```
Comment (ggwpez) [inline scripts/ci/gitlab/pipeline/test.yml:119]:
```suggestion
    - rustup install 1.70 && SKIP_WASM_BUILD=1 cargo +1.70 clippy --all-targets --locked -p polkadot-runtime-parachains
```

Comment (ggwpez) [reply scripts/ci/gitlab/pipeline/test.yml]: Obviously this is a shitty solution, but you now can debug from here why this is not working correctly in the docker without the `rustup`.

**Comment by alvicsam** [comment]
superseded by https://github.com/paritytech/polkadot/pull/7625

**Pull request closed without merging**